src/lib/extractor/Scanner.h

```diff
@@ -98,6 +102,27 @@ inline unsigned Scanner::nextPowerOfTwoPlusOne(unsigned v)
 	return std::max(3U, v + 2);
 }
 
+template <typename MAT>
+inline bool Scanner::will_it_scan(const MAT& unpadded_img)
+{
+	Scanner scanner(unpadded_img);
+	std::vector<Anchor> points = scanner.scan();
+	if (points.size() < 4)
+		return false;
+
+	constexpr int limit = 50;
+	Corners corners(points);
+	if (corners.top_left().x() > limit or corners.top_left().y() > limit)
```

Comment:
The check here is pretty simple. We run an extract on a freshly encoded image -- no blurring or skewing or any other camera-induced weirdness. We don't even need error correction for an image like this, it's pristine. The corners the scanner finds should be right where we expect them: at the edges of the image.

But if the (essentially random) data pattern is adversarial enough, the scanner will find a "corner" elsewhere, and we can detect/reject this frame as bad.